react/features/google-api/googleApi.native.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import {
+    API_URL_BROADCAST_STREAMS,
+    API_URL_LIVE_BROADCASTS
+} from './constants';
+
+/**
+ * Class to encapsulate Google API functionalities and provide a similar
+ * interface to what WEB has. The methods are different, but the point is that
+ * the export object is similar so no need for different export logic.
+ */
+class GoogleApi {
+
+    /**
+     * Retrieves the available YouTube streams the user can use for live
+     * streaming.
+     *
+     * @param {Object} googleAuth - The response from a successful Google Auth.
+     * @returns {Promise}
+     */
+    getYouTubeLiveStreams(googleAuth: Object): Promise<*> {
+        return new Promise((resolve, reject) => {
+
+            // Fetching the list of available broadcasts first.
+            this._fetchGoogleEndpoint(googleAuth, API_URL_LIVE_BROADCASTS)
```

Comment:
I'm not sure you need the promise wrapper if `this._fetchGoogleEndpoint` already returns a promise. I also think you can continue the chain below by returning `this._getLiveStreamsForBroadcasts`. So you'd be removing extra logic:
```
getYouTubeLiveStreams() {
  return this._fetchGoogleEndpoint(...)
    .then(broadcasts => this._getLiveStreamsForBroadcasts(...))
}
```
I think that's equivalent but I may be wrong. I think this comment may apple for the other methods below as well.

Reply:
I don't think that's equivalent. Created a quick JSFiddle to make sure we're on the same page: https://jsfiddle.net/4r8vdogf/7/

Is it what you meant?